src/test/java/in/nimbo/dao/EntryDAOTest.java

```diff
@@ -45,52 +59,124 @@ public static void init() throws SQLException, FileNotFoundException {
-                "DELETE FROM feed");
+                "DELETE FROM description;" + "DELETE FROM feed");
         statement.executeUpdate();
     }
 
-    @Test
-    public void save() throws SQLException {
-        descriptionDAO = PowerMockito.mock(DescriptionDAO.class);
-        contentDAO = PowerMockito.mock(ContentDAO.class);
-        EntryDAO entryDAO = new EntryDAOImpl(descriptionDAO, contentDAO);
+    private Entry createEntry(String title, Date pubDate, String content, SyndContent desc){
         Entry entry = new Entry();
-        List<Entry> entryList = new ArrayList<>();
         SyndEntry syndEntry = new SyndEntryImpl();
-        syndEntry.setPublishedDate(new Date());
-        entry.setSyndEntry(syndEntry);
-        if (!entryDAO.contain(entry)) {
-            entry = entryDAO.save(entry);
-        }
-        entryList.add(entry);
-        syndEntry = new SyndEntryImpl();
-        syndEntry.setPublishedDate(new Date());
-        syndEntry.setTitle("test");
-        entry = new Entry();
-        entry.setChannel("test");
         entry.setSyndEntry(syndEntry);
-        if (!entryDAO.contain(entry)) {
-            entry = entryDAO.save(entry);
+        syndEntry.setTitle(title);
+        syndEntry.setPublishedDate(pubDate);
+        syndEntry.setDescription(desc);
+        entry.setContent(content);
+        return entry;
+    }
+
+    @Test
+    public void save() throws SQLException {
+        List<Entry> entryList = new ArrayList<>();
+        for (int i = 0; i < 4; i++) {
+            SyndContent content = null;
+            if ((i & 2) != 0){
+                content = new SyndContentImpl();
+                content.setType("text/html");
+                content.setValue("desc " + i);
+            }
+            Entry entry = createEntry("title " + i, new Date(), (i & 1) != 0 ? "content " + i : null, content);
+            entry.getSyndEntry().setLink("link " + i);
+            if (!entryDAO.contain(entry)) {
+                entryDAO.save(entry);
+                entryList.add(entry);
+            }
         }
-        entryList.add(entry);
-        PreparedStatement statement = connection.prepareStatement("SELECT COUNT(*) as cnt FROM feed");
+        PreparedStatement statement = connection.prepareStatement("SELECT id FROM feed");
         ResultSet resultSet = statement.executeQuery();
-        if (resultSet.next()) {
-            int cnt = resultSet.getInt("cnt");
-            assertTrue(cnt > 0);
-            assertArrayEquals(entryList.toArray(), entryDAO.getEntries().toArray());
-        } else {
-            fail();
+        List<Entry> list = new ArrayList<>();
+        while (resultSet.next()){
+            Entry e = new Entry();
+            e.setId(resultSet.getInt("id"));
+            list.add(e);
         }
-        entryList.remove(0);
-        assertArrayEquals(entryList.toArray(), entryDAO.filterEntryByTitle(null, "e", null, null).toArray());
+        assertArrayEquals(entryList.toArray(), list.toArray());
```

Comment:
Change equals for Entry class

Reply:
Test will be failed because content and description of new entries are not same as before